### ISSUE TITLE
Push Messages don't arrive in production and when signed with adhoc profile

### ISSUE THREAD

**Issue by jwoyo** [FirebaseExtended/flutterfire#6025]
I'm using the newest version of firebase_messaging (9.1.3) with the newest flutter version (2.0.5). I integrated push messages for Apple as described in [the docs](https://firebase.flutter.dev/docs/messaging/apple-integration). Everything works fine on Android and when running "flutter run" on iOS. But after building and signing the app for production or with an adhoc provisioning profile no messages are received under iOS. 

Things I've tried:
- I assume the Firebase APNs Authentication Key to be set-up correctly as it works under Android and under iOS in development mode
- I assume that fiddling around with "FirebaseAppDelegateProxyEnabled" attribute in Info.plist is not needed anymore since firebase_messaging 9.x.x according to the migration guide.
- I checked that the provisioning profiles in use have "Push Notifications" under their "Enabled Capabilities" in Apple Developer portal.
- Even when using an adhoc signed build I'm able to obtain a device token. Sending a message to this token via firebase messaging SDK is successful. But the message never gets displayed on the device.

Does anyone have a clue what else to check? 

Cheers
Julian

**Comment by markusaksli-nc** [comment]
Hi @jwoyo 
Are you getting any error messages in the device logs? Are notification permissions granted and enabled on the device?
Thank you

**Comment by jwoyo** [comment]
I found the problem. For people stumble across this issue in the future:

Our ci environment overrid the file ios/Runner/Runner.entitlements which caused this xml nodes to be missing: 
`<key>aps-environment</key>
<string>production</string>`

I've been able to find this problem by using [ProvisionQL](https://github.com/ealeksandrov/ProvisionQL). This tool allowed me to inspect the *.ipa build files on my system. The line highlighted in red was missing in broken builds. 
![New Project (3)](https://user-images.githubusercontent.com/11311357/117290547-3abc8100-ae6e-11eb-9905-b9f95590a10b.png)

Another tip: Don't fiddle around with FirebaseAppDelegateProxyEnabled stuff in your plist files as recommended in many Stackoverflow topics. As far as I understand this is not needed anymore with firebase_messaging 9.x.x

Good luck
